src/data/mv_polynomial.lean

```diff
@@ -2,12 +2,72 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Johan Commelin, Mario Carneiro
-
-Multivariate Polynomial
 -/
 import algebra.ring
 import data.finsupp data.polynomial data.equiv.algebra
 
+/-!
+# Multivariate polynomials
+
+This file defines polynomial rings over a base ring (or even semiring),
+with variables corresponding to the terms in a general type σ (which could
+be infinite). It then establishes a substantial interface for this definition.
+
+## Important definitions
+
+In the definitions below, we use the following notation:
+`σ : Type*` (indexing the variables)
+`R : Type*` `[comm_semiring R]` (the coefficients)
+`s : σ →₀ ℕ`, with corresponding monomial written X^s
+`a : R`,
+`n : σ`, with corresponding monomial X_n
```

Comment:
```suggestion
`n : σ`, with corresponding monomial `X n`
```
The subscript comes out of nowhere here.

Reply:
It came from Johannes' docstring ;-)